### DISCUSSION TITLE
Flickering cursor in ssh domain pane

### DISCUSSION THREAD

**Discussion by neogeographica** [wezterm/wezterm#6217] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11, macOS

### Which Wayland compositor or X11 Window manager(s) are you using?

Mutter

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

When working in a pane that is in an ssh domain, sometimes I have seen the cursor flicker and go backward/forward while I'm typing, with the characters I'm typing also appearing to flicker as they appear. I can try to capture a video of this at some later date, but also I'm sure if you've seen this it will sound familiar.

I wondered if this was related to the predictive local echo... I don't think my connection is that bad, but when I was experimenting with showing some sort of display for is_tardy in pane metadata I noticed that turning on and off with some regularity. So, I cranked local_echo_threshold_ms up to 250 and this seeeeems to have gotten rid of the problem. Mostly. I don't know if it's coincidence, and also I've seen it again a couple of times since making the change.

Anyway, I just wanted to throw out a fishing line here to see if maybe this is some sort of widely known/understood behavior.

**Comment by loops** [comment]
Hi neogeographica,

What you're seeing is Wezterm preemptively updating the display, based on what is likely to happen.
It does this to mitigate against connection delays, in order to make the display feel more responsive.

In my experience, this is way more distracting than beneficial.   If you want to effectively disable this
feature you can set the delay inside each ssh domain to something high, so that it is never engaged:

For instance something like this inside each ssh configuration section:

    local_echo_threshold_ms = 20000,

**Comment by neogeographica** [comment]
Yah as mentioned above that's what I did. Hasn't completely solved the issue apparently, and I'm surprised if the connection really is that bad. But if that's the source of the behavior then there it is, I guess.

**Comment by loops** [comment, marked as answer]
Oops, i should have read more carefully.  But I think your setting of 1/4 of a second, might come into play once in a while.  I'd set it to something very high, and see if it solves the problem.  Don't think there is anything else that would be causing such flickering.

**Comment by neogeographica** [comment]
OK, thanks!
